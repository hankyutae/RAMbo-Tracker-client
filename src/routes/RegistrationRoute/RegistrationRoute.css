.registration-view {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  padding-top: 30px;
  max-width: 1092px;
}

.mission {
  text-align: left;
  width: 50%;
  align-self: center;
}


.reg-photo {
  max-width: 400px;
  border-radius: 0px 5px 5px 5px;
  /* border-top: 30px solid rgb(251, 177, 2); */
  box-shadow: 0px -30px 0 0 rgb(251, 177, 2);
  border-right: 30px solid rgb(251, 177, 2);
}

.reg-photo-container{
box-shadow: 5px 50px 20px -20px rgb(219, 219, 219);
margin-bottom: 50px;
}

.reg-photo-2 {
  margin-top: 10%;
  margin-bottom: 10%;
   max-width: 400px;
   box-shadow: -25px 25px 0 0 rgb(2, 11, 92);
}

.reg-description {
  margin-top: 0px;
  width: 90%;
}

.reg-description-header {
  margin-top: 30px;
  margin-bottom: 5px;
  width: 90%;
  font-size: 42px;
  font-family: cursive;
}

.emphasis {
  font-weight: bold;
  font-family: cursive;
  font-size: 28px;
}