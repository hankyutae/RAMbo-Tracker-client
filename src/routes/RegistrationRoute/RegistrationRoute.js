import React from 'react';
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';
import './RegistrationRoute.css'

class RegistrationRoute extends React.Component {
  handleRegistrationSuccess = () => {
    const { history } = this.props;
    history.push('/dash');
  }

  render() {
    return (
      <section aria-live="polite" className="registration-view">
        <RegistrationForm
        onRegistrationSuccess={this.handleRegistrationSuccess}
        />
        <img className="reg-photo" src="https://cdn-a.william-reed.com/var/wrbm_gb_food_pharma/storage/images/publications/food-beverage-nutrition/nutraingredients.com/news/markets-and-trends/savvy-shoppers-search-for-balance-is-protein-s-reign-over/8872621-1-eng-GB/Savvy-shoppers-search-for-balance-Is-protein-s-reign-over_wrbm_large.jpg" alt="Health food" />
        <h3 className="reg-description-header">Our mission</h3>
        <p className="reg-description">The RAMbo health tracker is an online food diary designed to help users take control of their health. By tracking foods they eat and symptoms they experience, Symptom Tracker will help you draw relationships between the food you eat and way you feel. By tracking your food and symptoms, you can now be better prepared to talk with your doctor about how to take your next steps to wellness. Happy tracking!</p>
      </section>
    )
  }
}

export default RegistrationRoute; 