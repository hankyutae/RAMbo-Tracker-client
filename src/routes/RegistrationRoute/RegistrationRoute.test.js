import React from "react";
import ReactDOM from "react-dom";
import RegistrationRoute from "./RegistrationRoute";
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer'

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <RegistrationRoute />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
it("renders the UI as expected", () => {
  const tree = renderer
    .create(<BrowserRouter>
      <RegistrationRoute />
    </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
