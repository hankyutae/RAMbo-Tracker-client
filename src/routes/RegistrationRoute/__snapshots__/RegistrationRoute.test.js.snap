// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  aria-live="polite"
  className="registration-view"
>
  <form
    className="registration-form"
    onSubmit={[Function]}
  >
    <h2>
      Register for an account
    </h2>
    <div
      className="error"
      role="alert"
    />
    <div
      className="registration-input"
    >
      <label
        className="regLabel"
        htmlFor="registration-name-input"
      >
        Enter your name:
      </label>
      <br />
      <input
        className="registration-input-field"
        id="registration-name-input"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="registration-input"
    >
      <label
        className="regLabel"
        htmlFor="registration-username-input"
      >
        Choose a username:
      </label>
      <br />
      <input
        className="registration-input-field"
        id="registration-username-input"
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="registration-input"
    >
      <label
        className="regLabel"
        htmlFor="registration-password-input1"
      >
        Choose a password:
      </label>
      <br />
      <input
        className="registration-input-field"
        id="registration-password-input1"
        name="password1"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="registration-input"
    >
      <label
        className="regLabel"
        htmlFor="registration-password-input2"
      >
        Re-enter your password:
      </label>
      <br />
      <input
        className="registration-input-field"
        id="registration-password-input2"
        name="password2"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="user-button"
      type="submit"
    >
      Sign up!
    </button>
    <p>
      <a
        href="/login"
        onClick={[Function]}
      >
        Already have an account?
      </a>
    </p>
  </form>
  <div
    className="reg-photo-container"
  >
    <img
      alt="A lemon"
      className="reg-photo"
      src="Lemons.jpg"
    />
  </div>
  <img
    alt="Carrot"
    className="reg-photo-2"
    src="carrot.jpg"
  />
  <div
    className="mission"
  >
    <h3
      className="reg-description-header"
    >
      Our mission
    </h3>
    <p
      className="reg-description"
    >
      The RAMbo health tracker is an online food diary designed to help you take control of your health.
       
    </p>
  </div>
  <div
    className="mission"
  >
    <p>
       
      <span
        className="emphasis"
      >
        By tracking foods
      </span>
       you eat and symptoms you experience, Symptom Tracker will help draw relationships between the what you eat and how you feel.
    </p>
  </div>
  <img
    alt="bunch of grapes"
    className="reg-photo-2"
    src="grapes.jpg"
  />
  <img
    alt="lettuce"
    className="reg-photo-2"
    src="lettuce.jpg"
  />
  <div
    className="mission"
  >
    <p>
       
      <span
        className="emphasis"
      >
        You can now
      </span>
       be better prepared to talk with your doctor about how to take your next steps to wellness. 
      <br />
       Happy tracking!
    </p>
  </div>
</section>
`;
